### PR TITLE
Call Class.forName() within doPrivileged block from WASURLObjectFactoryFinder

### PR DESCRIPTION
**Pull request by chackoj**

### CODE REVIEW

**Comment by chackoj** [comment]
#build

File: dev/com.ibm.ws.jndi/src/com/ibm/ws/jndi/internal/WASURLObjectFactoryFinder.java [hunk 1]
```diff
@@ -54,16 +53,26 @@ public ObjectFactory findFactory(String urlSchema, Hashtable<?, ?> env) throws N
         // Always add default prefix
         pkgPrefixes.add("com.sun.jndi.url");
 
+        try {
+            return findFactory(urlSchema, pkgPrefixes);
+        } catch (ClassNotFoundException e) {
+            // auto-ffdc
+        }
+    }
```
Comment (kristip17) [inline dev/com.ibm.ws.jndi/src/com/ibm/ws/jndi/internal/WASURLObjectFactoryFinder.java]: When I build locally with Java11, it made me add a return on findFactory (I added a return null at the end).

**Comment by kristip17** [comment]
I copied the changes locally so I can run the failing ldap_fat test (LDAPReferralTest). I had to make some changes to WASURLObjectFactoryFinder to get it to build -- I am building with Java11. Added review comments. After I made those changes, Java2 issue was resolved.

**Comment by chackoj** [comment]
#build

File: dev/com.ibm.ws.jndi/src/com/ibm/ws/jndi/internal/WASURLObjectFactoryFinder.java (hunk 1 above)
Comment (chackoj) [reply dev/com.ibm.ws.jndi/src/com/ibm/ws/jndi/internal/WASURLObjectFactoryFinder.java]: Thanks — an issue with my IDE prevented it flagging some errors. I found them in the build and resolved them.

**Pull request merged**